### PR TITLE
#FP-16 Admin registration page

### PR DESCRIPTION
**Pull request by stewartlloydjones**

### CODE REVIEW

File: src/main/resources/templates/Fragments/Forms.html [hunk 1]
```diff
@@ -305,4 +305,52 @@
...
+    <input type="password" id="password" class="form-control" name="password" placeholder="Password"  ><br>
+    <label id="confirmpasswordlbl" class="form-label" name="confirmpasswordlbl" >Confirm Password</label><br>
+    <input type="password" id="confirmpassword" class="form-control" name="confirmpassword" placeholder="Password"><br>
+    <label id="registrationcodelbl" class="form-label" name="registrationcodelbl" >Registration Code</label><br>
+    <input type="text" id="registrationcode" class="form-control" name="registrationcode"><br>
+    <button type="submit" class="btn btn-primary btn-search">Register Account!</button>
+
+
+    <script>
+        function validateAdminRegister() {
```
Comment (WillPaterson) [inline src/main/resources/templates/Fragments/Forms.html:327]: Not sure why but this doesn't properly validate the inputs. 
@Reason0x6 any ideas?

File: src/main/resources/templates/Fragments/Forms.html [hunk 2]
```diff
@@ -305,4 +305,52 @@
...
+
+            if (x == "") {
+                alert("Please fill in all fields");
+                return false;
+            }
+            else if(!(f["password"].value === f["confirmpassword"].value)){
+                alert("Passwords do not match");
+                return false;
+            }
+            else if(!(f["registrationcode"].value === "h38DxU9QneP1gB")){
```
Comment (WillPaterson) [inline src/main/resources/templates/Fragments/Forms.html:349]: This should probably be done server side

Comment (TobyChurches) [reply src/main/resources/templates/Fragments/Forms.html:349]: I think the 'registrationcode' should be verified on the server, Potentially in a different commit make the value generated by another admin account

File: src/main/resources/templates/Fragments/Forms.html [hunk 3]
```diff
@@ -305,4 +305,52 @@
...
+
+            if(f["email"].value == ""){ x = false; }
+            if(f["firstname"].value == ""){ x = false; }
+            if(f["lastname"].value == ""){ x = false; }
+            if(f["company"].value == ""){ x = false; }
+            if(f["password"].value == ""){ x = false; }
+            if(f["confirmpassword"].value == ""){ x = false; }
+            if(f["registrationcode"].value == ""){ x = false; }
+
+            if (x == "") {
```
Comment (TobyChurches) [inline src/main/resources/templates/Fragments/Forms.html:341]: Should this be testing if(x == false)?

Comment (WillPaterson) [reply src/main/resources/templates/Fragments/Forms.html:341]: The original code for user registration (Which I'm assuming this was based on. And that also validates correctly) has the same thing. I couldn't workout why this code wouldn't be working the same.

File: src/main/resources/templates/Fragments/Forms.html (hunk 1 above)
Comment (WillPaterson) [reply src/main/resources/templates/Fragments/Forms.html:327]: I pushed a fix to the branch for this @stewartlloydjones
Was just an issue with how the firstname and lastname had capital Names and the validation JS didn't.

**Review by coderuntheworld — APPROVED** [review]

**Review by TobyChurches — APPROVED** [review]

**Pull request merged**